packages/main/cypress/specs/TableSelection.cy.tsx

```diff
@@ -0,0 +1,374 @@
+		const selection = doc.getElementById("selection");
+
+		combobox?.addEventListener("change", () => {
+			if (selection && combobox) {
+				(selection as TableSelection).mode = TableSelectionMode[(combobox as ComboBox).value as keyof typeof TableSelectionMode];
+			}
+		});
+	});
+}
+
+describe("Mode - None", () => {
+	before(() => {
+		mountTestpage();
+		changeMode(TableSelectionMode.None);
+	});
+
+	it("selection should be not active", () => {
+		cy.get("#table0").as("table");
+		cy.get("#table0").children("ui5-table-header-row").as("headerRow");
+		cy.get("#table0").children("ui5-table-row").get("[row-key=\"0\"]").as("row0");
+		cy.get("#selection").as("selection");
+
+		cy.get("@table").should("exist");
+		cy.get("@headerRow").should("exist");
+		cy.get("@row0").should("exist");
+
+		cy.get("@selection").should("have.attr", "mode", TableSelectionMode.None);
+		cy.get("@headerRow").first().shadow().find("#selection-cell")
+			.should("not.exist");
+		cy.get("@row0").shadow().find("#selection-cell")
+			.should("not.exist");
+	});
+});
+
+const testConfig = {
+	"Single": {
+		"config": {
+			"mode": "Single",
+		},
+		"cases": {
+			"BOXES": {
+				"header": {
+					"exists": true,
+					"checkbox": false,
+				},
+				"row": {
+					"exists": true,
+					"checkbox": true,
+				},
+			},
+			"SPACE": {
+				"space_0": "0",
+				"space_4": "4",
+			},
+			"ARROWS_BOX": {
+				"arrow_initial": "0",
+				"arrow_down": "1",
+				"arrow_up": "0",
+			},
+			"MOUSE": {
+				"mouse_0": "0",
+				"mouse_4": "4",
+			},
+			"RANGE_MOUSE": {
+				"range_mouse_initial": "0",
+				"range_mouse_final": "4",
+				"range_mouse_edge": "0",
+			},
+			"RANGE_KEYBOARD": {
+				"initial": "0",
+				"block_1": "0",
+				"block_2": "6",
+			},
+		},
+	},
+	"Multiple": {
+		"config": {
+			"mode": "Multiple",
+		},
+		"cases": {
+			"BOXES": {
+				"header": {
+					"exists": true,
+					"checkbox": true
+				},
+				"row": {
+					"exists": true,
+					"checkbox": true
+				}
+			},
+			"SPACE": {
+				"space_0": "0",
+				"space_4": "0 4"
+			},
+			"ARROWS_BOX": {
+				"arrow_initial": "0",
+				"arrow_down": "0",
+				"arrow_up": "0"
+			},
+			"MOUSE": {
+				"mouse_0": "0",
+				"mouse_4": "0 4"
+			},
+			"RANGE_MOUSE": {
+				"range_mouse_initial": "0",
+				"range_mouse_final": "0 1 2 3 4",
+				"range_mouse_edge": "0 1 2 3 4"
+			},
+			"RANGE_KEYBOARD": {
+				"initial": "0",
+				"block_1": "0 1 2 3 4",
+				"block_2": "0 1 2 3 4 6 7 8 9"
+			}
+		}
+	}
+};
+
+Object.entries(testConfig).forEach(([mode, testConfigEntry]) => {
+	describe(`Mode - ${mode}`, () => {
+		beforeEach(() => {
+			clearSelection();
+			mountTestpage();
+			changeMode(testConfigEntry.config.mode as TableSelectionMode);
+		});
+
+		it("Correct boxes are shown", () => {
+			cy.get("#table0").as("table");
+			cy.get("#table0").children("ui5-table-header-row").as("headerRow");
+			cy.get("#table0").children("ui5-table-row").get("[row-key=\"0\"]").as("row0");
```

Comment:
this is repeating in every tests. you can move it to the beforeEach and remove the redundant code from other tests